src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala

```diff
@@ -63,14 +67,7 @@ object MethodRepository {
   }
 
   def validatePublicOrEmail(email:String): Boolean = {
-    val emailRegex = """^[a-zA-Z0-9\.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$""".r
-    email match {
-      case null => false
-      case x if x.trim.isEmpty => false
-      case x if emailRegex.findFirstMatchIn(x).isDefined => true
-      case x if x.equalsIgnoreCase("public") => true
-      case _ => false
-    }
+    "public".equals(email) || Try(new InternetAddress(email).validate()).isSuccess
```

Comment:
I'm glad we finally have email address validation, but I wasn't super psyched about maintaining our own regex, even if it was copied from a best-practice somewhere. What about this, using the javamail standard?